### PR TITLE
Fix conv and add convolve

### PR DESCRIPTION
**Pull request by SteveDiamond**
## Description
The ``conv`` operator was extremely broken, and gave incorrect shape information. This PR fixes bugs in ``conv``, deprecates ``conv``, and adds a ``convolve`` atom that matches np.convolve.

@PTNobel 

## Type of change
- [x] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [ ] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [x] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [x] Write unittests.
- [x] Run the unittests and check that they’re passing.
- [x] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

File: cvxpy/atoms/affine/conv.py [hunk 1]
```diff
@@ -69,9 +72,94 @@ def validate_arguments(self) -> None:
...
+
+    def sign_from_args(self) -> Tuple[bool, bool]:
+        """Same as times.
+        """
+        return u.sign.mul_sign(self.args[0], self.args[1])
+
+    def is_incr(self, idx) -> bool:
+        """Is the composition non-decreasing in argument idx?
+        """
+        return self.args[0].is_nonneg()
```
Comment (PTNobel) [inline cvxpy/atoms/affine/conv.py:91]: Why do you not need idx? Shouldn't it be `self.args[0][idx].is_nonneg()`?

File: cvxpy/atoms/affine/conv.py [hunk 2]
```diff
@@ -69,9 +72,94 @@ def validate_arguments(self) -> None:
...
+
+    def is_decr(self, idx) -> bool:
+        """Is the composition non-increasing in argument idx?
+        """
+        return self.args[0].is_nonpos()
```
Comment (PTNobel) [inline cvxpy/atoms/affine/conv.py:96]: Ditto above.

File: cvxpy/atoms/affine/conv.py [hunk 3]
```diff
@@ -69,9 +72,94 @@ def validate_arguments(self) -> None:
...
+            raise ValueError("The first argument to conv must be constant.")
+
+    def shape_from_args(self) -> Tuple[int, int]:
+        """The sum of the argument dimensions - 1.
+        """
+        lh_length = self.args[0].size
+        rh_length = self.args[1].size
+        return (lh_length + rh_length - 1,)
 
     def sign_from_args(self) -> Tuple[bool, bool]:
```
Comment (PTNobel) [inline cvxpy/atoms/affine/conv.py:164]: I think for the copied methods above, my questions from above might need to be copied below too.

**Review by PTNobel — COMMENTED** [review]
I didn't look too closely at the tests. They made sense to me, but I can check them more carefully if we're worried they may be missing edge cases.

File: cvxpy/atoms/affine/conv.py (hunk 1 above)
Comment (SteveDiamond) [reply cvxpy/atoms/affine/conv.py:91]: If idx were used, it would be ``self.args[idx].is_nonneg()``. But this atom requires that the first argument be constant, so it's always the first argument that is checked.

Comment (PTNobel) [reply cvxpy/atoms/affine/conv.py:91]: Ah okay; I had misunderstood the function. Looks good to me then!

**Comment by SteveDiamond** [comment]
Thanks @PTNobel for the quick turn around!
